src/inmanta/module.py

```diff
@@ -1242,19 +1291,32 @@ def get_name_from_metadata(cls, metadata: T) -> str:
         raise NotImplementedError()
 
     @abstractmethod
-    def add_module_requirement(self, requirement: InmantaModuleRequirement, add_as_v1_module: bool) -> None:
+    def add_module_requirement_persistent(self, requirement: InmantaModuleRequirement, add_as_v1_module: bool) -> None:
         """
-        Add a new module requirement to the meta-data.
+        Add a new module requirement to the files that define requirements on other modules. This could include the
+        requirements.txt file next to the metadata file of the project or module. This method updates the files on disk.
+
+        This operation may make this object invalid or outdated.
```

Comment:
What exactly do you mean by this? The metadata is refreshed afterward, isn't it?

Reply:
I see this is in response to @wouterdb's remark that reloading the metadata after construction could leave the object inconsistent because it's used for initialization. Could we try to make this a bit more strict? For example "This operation may make this object invalid or outdated if the persisted metadata has been updated since creating this instance."?

Reply:
I updated to docstring